noun/manage.c

```diff
@@ -1180,28 +1180,40 @@ u3m_soft(c3_w    sec_w,
 
   if ( 0 == u3h(why) ) {
     return why;
-  } else {
-    u3_noun tax, cod, pro, mok;
+  }
+  else {
+    //  don't use .^ at the top level!
+    //
+    c3_assert(1 != u3h(why));
 
-    c3_assert(1 != u3h(why));  //  don't use .^ at the top level!
-    
-    if ( 2 == u3h(why) ) {
-      cod = c3__exit;
-      tax = u3k(u3t(why));
-    } 
+    //  don't call +mook if we have no kernel
+    //
+    if ( 0 == u3A->roc ) {
+      u3z(why);
+      return u3nc(2, u3_nul);
```

Comment:
On second thought, this doesn't seem to be entirely consistent with the contract of `u3m_soft`. In the normal case, errors are produced with a specific mote instead of the original numeric tag. The success case is still consistent, and these changes are only being made to support soft-ing without an installed kernel, which is a rare case. I'm comfortable with the inconsistency, but i'll leave a comment explaining it further.